test/evp_test.c

```diff
@@ -1618,11 +1621,24 @@ static int mac_test_run_mac(EVP_TEST *t)
         t->err = "MAC_CREATE_ERROR";
         goto err;
     }
-
+    /* get mac-size before init */
+    size = EVP_MAC_CTX_get_mac_size(ctx);
     if (!EVP_MAC_init(ctx, expected->key, expected->key_len, params)) {
         t->err = "MAC_INIT_ERROR";
         goto err;
     }
+    if (size != 0) {
```

Comment:
Not sure I understand this. Don't we expect that `EVP_MAC_CTX_get_mac_size()` will return 0 if called before init (at least for cmac)? In which case we will never run this block because `size == 0`? What am I missing?

Reply:
We need a test that (at least) ensures we no longer seg-fault when `EVP_MAC_CTX_get_mac_size()` is called before init.  The call I added at line 1625 does that.

If you remove my change to `evp_lib.c`, re-compile and run this test, then it will fail.

The additional checks (starting at line 1630) ensure that the value returned by `EVP_MAC_CTX_get_mac_size()` is consistent.  But we can only do them if `size != 0` (b/c the docs say if 0 is returned, then calling init will set the mac size).

Reply:
Perhaps adding a further check that fails the test when `size_before == 0 && size_after == 0` makes sense.